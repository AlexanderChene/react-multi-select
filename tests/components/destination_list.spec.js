import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";

import DestinationList from "../../src/components/destination_list";
import List from "../../src/components/list/items_list";

const CustomComponent = jest
  .fn(() => <div>Custom Component</div>)
  .mockName("mockedComponent");

const custom_messages = {
  searchPlaceholder: "Find...",
  noItemsMessage: "No entries available...",
  noneSelectedMessage: "Nothing",
  selectedMessage: "Checked",
  selectAllMessage: "Check all",
  clearAllMessage: "Uncheck all"
};

const unselectItems = jest.fn().mockName("unselectItems");
const clearAll = jest.fn().mockName("clearAll");

describe("DestinationList", () => {
  test("default snapshot", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<DestinationList />);
    expect(tree).toMatchSnapshot();
  });

  test("custom selectionStatusRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList selectionStatusRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom selectedItemRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList selectedItemRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom noItemsRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList noItemsRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom messages", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList messages={custom_messages} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed unselectItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList unselectItems={unselectItems} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed clearAll", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<DestinationList clearAll={clearAll} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedIds", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<DestinationList selectedIds={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<DestinationList selectedItems={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedItems", () => {
    const onClick = jest.fn();
    const component = shallow(<DestinationList unselectItems={onClick} />);
    const list = component.find(List).at(0);
    list.simulate("click", {});
    expect(onClick).toHaveBeenCalledWith([{}]);
  });
});
