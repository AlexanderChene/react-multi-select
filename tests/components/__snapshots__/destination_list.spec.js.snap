// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationList custom messages 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Uncheck all"
    noneSelectedMessage="Nothing"
    selected={Array []}
    selectedMessage="Checked"
  />
  <ItemsList
    height={355}
    itemHeight={40}
    items={Array []}
    noItemsMessage="No entries available..."
    noItemsRenderer={[Function]}
    offset={0}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`DestinationList custom noItemsRenderer 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={Array []}
    selectedMessage="selected"
  />
  <ItemsList
    height={355}
    itemHeight={40}
    items={Array []}
    noItemsMessage={undefined}
    noItemsRenderer={[MockFunction mockedComponent]}
    offset={0}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`DestinationList custom selectedItemRenderer 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={Array []}
    selectedMessage="selected"
  />
  <ItemsList
    height={355}
    itemHeight={40}
    items={Array []}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={undefined}
    renderer={[MockFunction mockedComponent]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`DestinationList custom selectionStatusRenderer 1`] = `
<Column>
  <mockConstructor
    clearAll={undefined}
    clearAllMessage={undefined}
    noneSelectedMessage={undefined}
    selected={Array []}
    selectedMessage={undefined}
  />
  <ItemsList
    height={355}
    itemHeight={40}
    items={Array []}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`DestinationList default snapshot 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={Array []}
    selectedMessage="selected"
  />
  <ItemsList
    height={355}
    itemHeight={40}
    items={Array []}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`DestinationList passed clearAll 1`] = `
<Column>
  <SelectionStatus
    clearAll={[MockFunction clearAll]}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={Array []}
    selectedMessage="selected"
  />
  <ItemsList
    height={355}
    itemHeight={40}
    items={Array []}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`DestinationList passed selectedIds 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={
      Array [
        1,
        2,
      ]
    }
    selectedMessage="selected"
  />
  <ItemsList
    height={355}
    itemHeight={40}
    items={Array []}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`DestinationList passed selectedItems 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={Array []}
    selectedMessage="selected"
  />
  <ItemsList
    height={355}
    itemHeight={40}
    items={
      Array [
        1,
        2,
      ]
    }
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={undefined}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;

exports[`DestinationList passed unselectItem 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={Array []}
    selectedMessage="selected"
  />
  <ItemsList
    height={355}
    itemHeight={40}
    items={Array []}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[MockFunction unselectItem]}
    renderer={[Function]}
    selectedIds={Array []}
  />
</Column>
`;
